tests/test_estimation.py

```diff
@@ -384,6 +383,28 @@ def test_conditional_expectation_returns_same_value_as_Hardie_excel_sheet(self,
         actual = bfg.conditional_expected_number_of_purchases_up_to_time(t, x, t_x, T)
         assert abs(expected - actual) < 0.001
 
+    def test_conditional_expectation_with_negative_hyp2f1_term(self, cdnow_customers):
+        bfg = estimation.BetaGeoFitter()
+        bfg.params_ = OrderedDict([('r', 0.5458741247391189), ('alpha', 13.409316394557274), ('a', 0.0009994943799344323), ('b', 0.03899022143378801)])
+        t = 180
+        x = 0
+        t_x = 0
+        T = 5
+        expected = 5.212
+        actual = bfg.conditional_expected_number_of_purchases_up_to_time(t, x, t_x, T)
+        assert abs(expected - actual) < 0.001
+
+    def test_conditional_expectation_overflow_error_with_high_frequency(self, cdnow_customers):
```

Comment:
this test is failing in CI: https://travis-ci.org/CamDavidsonPilon/lifetimes/jobs/429049394#L705

does it pass locally?

Reply:
Yes, @CamDavidsonPilon, thanks for notifying! So essentially I realised this patch will not pass when the frequency has some high values as the`hyp2f1_term` becomes `inf`. In the original patch, the `log` trick does solve the issue, but won't accept a negative `hyp2f1_term`. So now I am stuck in this predicament. 

Any ideas how we can solve both of these problems?

Reply:
Looks like `mpmath.hyp2f1` fixes the overflow issue encountered when using `scipy.specia.hyp2f1`... Interesting....